src/dataset/create_dataset.py

```diff
@@ -19,240 +20,141 @@
-        
-        self.sources_so_far: List[str] = []
-        self.info_types: Dict[str, List[str]] = {}
-    
-    def extract_info_from_article(self, article: Dict[str, Any]) -> Tuple[str]:
-        """
-        This function extracts the title, author, date, URL, tags, and text from an article.
-        
-        Args:
-            article (Dict[str, Any]): a dictionary containing the article's text and metadata.
-
-        Returns:
-            Tuple[str]: a tuple containing the title, author, date, URL, tags, and text of the article.
-        """
-        title: str = ""
-        author: str = ""
-        date_published: str = None
-        url: str = None
-        tags: str = None
-        text: str = None
-        
-        # Get title
-        if 'title' in article and 'book_title' in article and article['title']: title = article['title']
-        elif 'book_title' in article and 'title' not in article and article['book_title']: 
-            title = article['book_title']
-        elif 'title' in article and article['title']: 
-            title = article['title']
-        title = title.strip('\n').replace('\n', ' ')[:100]
-
-        # Get author
-        if 'author' in article and 'authors' in article and article['author']: author = article['author']
-        elif 'authors' in article and article['authors']: author = article['authors']
-        elif 'author' in article and article['author']: author = article['author']
-        if type(author) == str: author = get_authors_list(author)
-        if type(author) == list: author = ', '.join(author)
-        author = author.strip('\n').replace('\n', ' ')[:100]
-
-        # Get date published
-        if 'date_published' in article and article['date_published'] and len(article['date_published']) >= 10: date_published = article['date_published'][:10]
-        elif 'published' in article and article['published'] and len(article['published']) >= 16: date_published = article['published'][:16]
-        else: date_published = None
-        if date_published is not None:
-            date_published = standardize_date(date_published)
-            
-        # Get URL
-        if 'link' in article and article['link']: url = article['link']
-        elif 'url' in article and article['url']: url = article['url']
-        elif 'doi' in article and article['doi']: url = article['doi']
-        else: url = None
-            
-        # Get tags
-        if 'tags' in article and article['tags']:
-            if type(article['tags']) == list: tags = ', '.join([val['term'] for val in article['tags']])
-            elif type(article['tags']) == str: tags = article['tags']
-            else: tags = None
+                self.entries_per_source_count[source] = 0
         
-        # Get text
-        if 'text' in article and article['text']: text = article['text']
-        else:
-            raise MissingDataException(f"Entry has no text.")
+    def contains_required_metadata(self, entry: Dict[str, Any]):
```

Comment:
[NIT] maybe `validate_required_metadata`? I have the feeling that function that are named like `contains_`, `is_` etc. tend to return a boolean, while functions which will raise an exception if something is wrong tend to be called `validate_`

Reply:
Good idea. Since looking at the ARD code yesterday I also learned about staticmethod and should this not be a case where I call it like 
@staticmethod
def _validate_required_metadata, since I wouldn't want someone to use this function outside of the class since that wouldn't make much sense?

Reply:
and because it doesn't use self. *

Reply:
yup, a staticmethod makes sense. It's not really an issue - I personally like to use them as I try to keep functions as pure as possible and that helps mark them are purer. I also prefer to use raw functions, rather than classes, but that's a different issue...
`_validate_required_metadata` makes sense, yes. Don't worry too much about naming things though. After all - it's one of the [two hard things in CS](https://twitter.com/secretGeek/status/7269997868?ref_src=twsrc%5Etfw%7Ctwcamp%5Etweetembed%7Ctwterm%5E7269997868%7Ctwgr%5E279ff3998e5bf3ccaa3d6723d084070236fa1cf3%7Ctwcon%5Es1_&ref_url=https%3A%2F%2Fwww.martinfowler.com%2Fbliki%2FTwoHardThings.html)